algorithms/perf_test/test_inclusive_scan.cpp

```diff
@@ -0,0 +1,143 @@
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdio>
+#include <cstdint>
+#include <cstring>
+#include <cstdlib>
+#include <random>
+
+#include <benchmark/benchmark.h>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Timer.hpp>
+#include <Kokkos_StdAlgorithms.hpp>
+
+#include "../../core/perf_test/Benchmark_Context.hpp"
+
+namespace KE = Kokkos::Experimental;
+
+using exec_space = Kokkos::DefaultExecutionSpace;
+
+template <class ValueType>
+struct SumFunctor {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const ValueType& a, const ValueType& b) const {
+    return (a + b);
+  }
+};
+
+template <class ValueType>
+struct MaxFunctor {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const ValueType& a, const ValueType& b) const {
+    if (a > b)
+      return a;
+    else
+      return b;
+  }
+};
+
+// Helper to obtain last element of a view
+template <class T>
+T obtainLastElem(const Kokkos::View<T*, exec_space>& out) {
+  auto h_out = Kokkos::create_mirror_view(out);
+  Kokkos::deep_copy(h_out, out);
+  auto view_size = h_out.extent(0);
+  return h_out(view_size - 1);
+}
+
+template <class T>
+void InclusiveScan(benchmark::State& state,
+                   const Kokkos::View<T*, exec_space>& in,
+                   Kokkos::View<T*, exec_space>& out) {
+  // Default prefix sum scan
+  Kokkos::Timer timer;
+  KE::inclusive_scan("default scan", exec_space(), KE::cbegin(in), KE::cend(in),
+                     KE::begin(out));
+  exec_space().fence();
+  double time_def = timer.seconds();
+
+  T res_def = obtainLastElem(out);
+
+  // Prefix sum scan using a sum functor
+  SumFunctor<T> sum_functor;
+
+  timer.reset();
+  KE::inclusive_scan("sum functor scan", exec_space(), KE::cbegin(in),
+                     KE::cend(in), KE::begin(out), sum_functor);
+  exec_space().fence();
+  double time_sum_func = timer.seconds();
+
+  T res_sum_func = obtainLastElem(out);
+
+  // Scan using a max functor
+  MaxFunctor<T> max_functor;
+
+  timer.reset();
+  KE::inclusive_scan("max functor scan", exec_space(), KE::cbegin(in),
+                     KE::cend(in), KE::begin(out), max_functor);
+  exec_space().fence();
+  double time_max_func = timer.seconds();
+
+  // Checking expected results
+  std::size_t res_max_func = static_cast<std::size_t>(obtainLastElem(out));
+  std::size_t exp_max_val  = in.size() - 1;
+  bool passed = (res_def == res_sum_func) && (res_max_func == exp_max_val);
+
+  state.counters["Passed"]             = benchmark::Counter(passed);
+  state.counters["Value default scan"] = benchmark::Counter(res_def);
+  state.counters["Value sum functor"]  = benchmark::Counter(res_sum_func);
+  state.counters["Value max functor"]  = benchmark::Counter(res_max_func);
```

Comment:
This feels out-of-place. I would rather error out if the results aren't correct.